tests/benchmarks/FCSBenchmarks/CompilerServiceBenchmarks/FileCascadeBenchmarks.fs

```diff
@@ -0,0 +1,118 @@
+﻿namespace FSharp.Compiler.Benchmarks
+
+open System
+open System.IO
+open System.Text
+open FSharp.Compiler.CodeAnalysis
+open FSharp.Compiler.Diagnostics
+open FSharp.Compiler.EditorServices
+open FSharp.Compiler.Text
+open FSharp.Compiler.AbstractIL.IL
+open FSharp.Compiler.AbstractIL.ILBinaryReader
+open BenchmarkDotNet.Attributes
+open FSharp.Compiler.Benchmarks
+open Microsoft.CodeAnalysis.Text
+open BenchmarkDotNet.Order
+open BenchmarkDotNet.Mathematics
+
+[<AutoOpen>]
+module private CascadeProjectHelpers =
+        
+    let createProject files projectFilename =       
+        {
+            ProjectFileName = projectFilename
+            ProjectId = None
+            SourceFiles = files
+            OtherOptions =  [|"--optimize+" |] 
```

Comment:
You can get good starting options like this:

```fsharp
let checker = FSharpChecker.Create(projectCacheSize=300)

let projectOptions, _diagnostics =
    checker.GetProjectOptionsFromScript("file.fs", SourceText.ofString "", assumeDotNetFramework=false)
    |> Async.RunSynchronously
``` 

And then overwrite what you need. If I remember correctly this worked for me on Mac. 
https://github.com/0101/performance/blob/ff3c20b8729f1c9ff57936e0fe31a1284935b5d8/src/benchmarks/real-world/FSharp/Program.fs#L41-L56

Reply:
@nojaf : I cannot test this on a Mac myself. For WSL, I can have a look if that works out of the box or if more commands are needed.
I am also wondering what might be causing the difference between the needs across OSes (?).

Reply:
> I am also wondering what might be causing the difference between the needs across OSes (?).

Likely different SDK references, maybe differences in resolvers.

Reply:
@vzarytovskii this wasn't working for me some time ago:
https://github.com/nojaf/FSharpProjectRepro/actions/runs/3240969667/jobs/5312280460
Notice how the project options are the same: https://github.com/nojaf/FSharpProjectRepro/blob/a7f903e4a5e980ff8b949ebf9d8fa2d3cec0a568/UnitTest1.fs#L21-L34

In hindsight, I do see that mixing `FSharp.Compiler.Service, Version=41.0.6` and `dotnet 7 rc 1` may not have been the best combination.
Updating to the latest compiler also doesn't work because `Microsoft.Build.Framework >= 17.4.0-preview-22469-04` is not on the public NuGet feed. (That is a different story).

Is there any more documentation on why the F# compiler really needs to know about all these references to compile something like:
```fsharp
module Benchmark0
let returnValue = 5
let myFunc0 () = 5
```
?

That part just really bites me so often...